src/context.rs

```diff
@@ -203,107 +206,67 @@ impl<A: Actor> Context<A> {
     pub async fn run(mut self, mut actor: A) -> A::Stop {
         actor.started(&mut self).await;
 
-        // Idk why anyone would do this, but we have to check that they didn't do ctx.stop()
-        // in the started method, otherwise it would kinda be a bug
         if !self.check_running(&mut actor).await {
-            self.stop_all();
             return actor.stopped().await;
         }
 
-        // Similar to above
-        if let Some(BroadcastMessage::Shutdown) = self.broadcast_receiver.try_recv().unwrap() {
-            return actor.stopped().await;
-        }
+        // // Similar to above
+        // if let Some(BroadcastMessage::Shutdown) = self.broadcast_receiver.try_recv().unwrap() {
```

Comment:
I've just realised that there is a bug on master relating to this - it will swallow a `Message` variant from other actors on the same address, so that extra case should probably be handled. Regardless of that, I don't think this should go, in case of this actor being attached to the context late. Imagine:

Actor A starts
Actor A spawns actor B and spawns it onto an executor
Actor A broadcasts shutdown
Actor B's task is polled for the first time, and if this code isn't here, it will not shut down properly